clients/docs/java.rst

```diff
@@ -0,0 +1,317 @@
+      [Consumed record]: alice, 3
+      [Consumed record]: alice, 4
+      [Consumed record]: alice, 5
+      [Consumed record]: alice, 6
+      [Consumed record]: alice, 7
+      [Consumed record]: alice, 8
+      [Consumed record]: alice, 9
+      ...
+      [Running count]: alice, 0
+      [Running count]: alice, 1
+      [Running count]: alice, 3
+      [Running count]: alice, 6
+      [Running count]: alice, 10
+      [Running count]: alice, 15
+      [Running count]: alice, 21
+      [Running count]: alice, 28
+      [Running count]: alice, 36
+      [Running count]: alice, 45
+      ...
+
+#. When you are done, press ``Ctrl-C``.
+
+#. View the :devx-examples:`Kafka Streams code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/StreamsExample.java`.
+
+
+Avro and Confluent Cloud Schema Registry
+-----------------------------------------
+
+This example is similar to the previous example, except the value is formatted
+as Avro and integrates with the |ccloud| |sr|. Before using |ccloud| |sr|, check
+its |ccloud| |sr|.
+
+#. .. include:: includes/client-example-vpc.rst
+
+#. .. include:: includes/client-example-schema-registry-1.rst
+
+#. .. include:: includes/client-example-schema-registry-2.rst
+
+#. .. include:: includes/schema-registry-java.rst
+
+
+Produce Avro Records
+~~~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
+      mvn clean package
+
+      # Run the Avro producer
+        # If the topic does not already exist, the code will use the Kafka Admin Client API to create the topic
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.ProducerAvroExample" \
+      -Dexec.args="$HOME/.confluent/java.config test2"
+
+#. View the :devx-examples:`producer Avro code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ProducerAvroExample.java`.
+
+
+Consume Avro Records
+~~~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
+      mvn clean package
+
+      # Run the Avro consumer
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.ConsumerAvroExample" \
+      -Dexec.args="$HOME/.confluent/java.config test2"
+
+#. View the :devx-examples:`consumer Avro code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java`.
+
+
+Avro Kafka Streams
+~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro Kafka Streams application, passing in arguments for:
+
+   -  the local file with configuration parameters to connect to your Kafka
+      cluster
+   -  the same topic name you used earlier
+
+   .. code-block:: text
+
+      # Compile the Java code
```

Comment:
Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)